File: ios/StockAgents/Models/Verdict.swift
import Foundation

/// The five-tier Portfolio Manager rating, ordered bearish → bullish. Ported
/// from `RATING_POS` / `RATING_COLOR` in `web/static/index.html` and
/// `RATING_ORDER` in `tradingagents/brokers/intents.py`.
enum Verdict: String, CaseIterable, Equatable {
    case sell = "Sell"
    case underweight = "Underweight"
    case hold = "Hold"
    case overweight = "Overweight"
    case buy = "Buy"

    /// Tolerant parse of the backend `decision` string (case-insensitive).
    /// Unknown values map to `nil` so callers can show the raw word neutrally.
    init?(decision: String) {
        let trimmed = decision.trimmingCharacters(in: .whitespacesAndNewlines)
        guard let match = Verdict.allCases.first(where: {
            $0.rawValue.caseInsensitiveCompare(trimmed) == .orderedSame
        }) else { return nil }
        self = match
    }

    /// Marker position on the Sell→Buy gauge, 0–100 (`RATING_POS`).
    var gaugePosition: Double {
        switch self {
        case .sell: return 0
        case .underweight: return 25
        case .hold: return 50
        case .overweight: return 75
        case .buy: return 100
        }
    }

    /// Accent color hex (`RATING_COLOR`).
    var colorHex: String {
        switch self {
        case .sell: return "#f43f5e"
        case .underweight: return "#fb923c"
        case .hold: return "#94a3b8"
        case .overweight: return "#4ade80"
        case .buy: return "#10b981"
        }
    }
}

File: mobile-rn/src/models/verdict.ts
/**
 * The five-tier Portfolio Manager rating, ordered bearish → bullish. Ported from
 * `RATING_POS` / `RATING_COLOR` in `web/static/index.html` and `RATING_ORDER` in
 * `tradingagents/brokers/intents.py`.
 */
export type Verdict = 'Sell' | 'Underweight' | 'Hold' | 'Overweight' | 'Buy';

export const VERDICT_ORDER: Verdict[] = ['Sell', 'Underweight', 'Hold', 'Overweight', 'Buy'];

/** Marker position on the Sell→Buy gauge, 0–100 (`RATING_POS`). */
export const VERDICT_GAUGE_POSITION: Record<Verdict, number> = {
  Sell: 0,
  Underweight: 25,
  Hold: 50,
  Overweight: 75,
  Buy: 100,
};

/** Accent color hex per tier (`RATING_COLOR`). */
export const VERDICT_COLOR: Record<Verdict, string> = {
  Sell: '#f43f5e',
  Underweight: '#fb923c',
  Hold: '#94a3b8',
  Overweight: '#4ade80',
  Buy: '#10b981',
};

/** Sell→Buy gradient stops used by the gauge bar. */
export const VERDICT_GRADIENT = VERDICT_ORDER.map((v) => VERDICT_COLOR[v]);

/**
 * Tolerant parse of the backend `decision` string (case-insensitive). Unknown
 * values map to `null` so callers can show the raw word neutrally.
 */
export function parseVerdict(decision: string): Verdict | null {
  const trimmed = decision.trim().toLowerCase();
  return VERDICT_ORDER.find((v) => v.toLowerCase() === trimmed) ?? null;
}
